src/metadata/metadata.c

```diff
@@ -1192,8 +1192,9 @@ static void _recovery_rebuild_cline_metadata(ocf_cache_t cache,
 		env_atomic_inc(&core->runtime_meta->dirty_clines);
 		env_atomic_inc(&core->runtime_meta->
 				part_counters[part_id].dirty_clines);
-		env_atomic64_cmpxchg(&core->runtime_meta->dirty_since,
-				0, env_get_tick_count());
+		if (!env_atomic64_read(&core->runtime_meta->dirty_since))
+			env_atomic64_xchg(&core->runtime_meta->dirty_since,
```

Comment:
I think the previous version (env_atomic64_cmpxchg) was more precise

Reply:
Then double check would be done - first we check it and then `atomic_cmpxch` would do check. Wouldn't it be redundant?

Reply:
I think it would be ok.  First of all the double check would be relatively rare, it happens only when dirty counter is incremented from zero. Secondly the proper way of not double-checking would be to do it the old way (only env_atomic64_cmpxchg without any condition) - but this might have a real perf impact every I/O. I believe the double check that I suggest would be a good compromise between performance and precision (there would be no time-of-check vs time of use mismatch), although the difference between this and the current version is probably negligible. You know my opinion, please use your judgement :).

Reply:
Then the new wrapper in the CAS for the atomic function would be unnecessary and IMHO less new code is better. Thanks for that explanation.